lib/ProductOpener/Ingredients.pm

```diff
@@ -3371,20 +3371,47 @@ sub get_missing_ecobalyse_ids ($ingredients_ref) {
 		# Getting properties from product
 		#my $product_origin = has_specific_ingredient_property()
 
-		# Getting the correct ecobalyse comake devde
-		my $ecobalyse_code = get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse:en");
-		if (defined $ecobalyse_code) {
-			$ingredient_ref->{ecobalyse_code} = $ecobalyse_code;
+		# If the ingredient is organic, the 'ecobalyse_labels_en_organic:en' id is used
+		if ($ingredient_ref->{labels} =~ /\ben:organic\b/) {
```

Comment:
This approach won't work if we have an organic ingredient for which we have a general ecobalyse match, but no organic ecobalyse match.

e.g. if we have "organic pear", in the ingredients, we don't have it in Ecobalyse:

ecobalyse:en: pear-eu
ecobalyse_origins_en_france:en: pear-fr

but we need to fallback on pear-eu.

So I suggest a different approach:

Based on the labels, origins etc. of the product, construct a list of prioritized ecobalyse suffixes.

e.g. if the product is "organic pear from france", we might want to get:

1. ecobalyse_origins_france_label_organic
2. ecobalyse_origins_european-union_label_organic
3. ecobalyse_label_organic
4. ecobalyse_origins_france
5. ecobalyse_origins_european-uinion
6. ecobalyse

(or maybe a different order, depending on what we consider more important: the origin or the label)

Then we check if we have those as ecobalyse properties for "en:pear", and we stop at the first one we find.

Reply:
Good point indeed, i'll change my code consequently